pkg/services/ngalert/notifier/alertmanager_config.go

```diff
@@ -181,14 +182,60 @@ func (moa *MultiOrgAlertmanager) ApplyAlertmanagerConfiguration(ctx context.Cont
 		}
 	}
 
-	if err := am.SaveAndApplyConfig(ctx, &config); err != nil {
+	if err := moa.SaveAndApplyConfig(ctx, org, am, &config); err != nil {
 		moa.logger.Error("Unable to save and apply alertmanager configuration", "error", err)
 		return AlertmanagerConfigRejectedError{err}
 	}
 
 	return nil
 }
 
+// SaveAndApplyConfig saves the given configuration to the database and applies it to the given alertmanager in a single transaction.
```

Comment:
This is a single database transaction? If the transaction aborts on commit in the database, will the configuration still be applied to the Alertmanager? Do we need to roll it back on abort?

Reply:
If the transaction aborts on commit, the AM config will temporarily be out-of-sync with the database, correct. Assuming the issue is transient, this should fix itself on a subsequent periodic sync (every minute).

We could consider rolling back the AM config too, but considering that's how it currently works as well, I'd prefer to leave that for its own PR to keep this one targeted.